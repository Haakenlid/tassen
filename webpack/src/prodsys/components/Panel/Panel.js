import SentryBoundary from 'common/components/SentryBoundary'
import cx from 'classnames'

const TopBar = ({ children }) => <nav className="TopBar">{children}</nav>
const BottomBar = ({ children }) => <nav className="BottomBar">{children}</nav>

const Panel = ({
  children,
  header = null,
  footer = null,
  className,
  scroll = true,
  ...props
}) => (
  <section className={cx('Panel', className)} {...props}>
    <SentryBoundary>
      {header && <TopBar>{header}</TopBar>}
      <SentryBoundary>
        <section className={cx('content', { scroll })}>{children}</section>
      </SentryBoundary>
      {footer && <BottomBar>{footer}</BottomBar>}
    </SentryBoundary>
  </section>
)
export default Panel
