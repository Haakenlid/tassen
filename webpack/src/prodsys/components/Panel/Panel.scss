@import 'styles/globals.scss';
%button {
  border-radius: 0.3rem;
  box-shadow: inset -1px -1px 1px rgba(black, 0.4),
    inset 1px 1px 1px rgba(white, 0.3);
}

.Panel {
  border-left: 1px solid #777;
  flex: 1;
  align-self: flex-start;
  min-height: 0;
  height: 100vh;
  display: flex;
  flex-direction: column;
  .content {
    flex: 1;
    display: flex;
    flex-direction: column;
    height: 0;
    margin: 0;
    &.scroll {
      overflow-y: scroll;
      overflow-x: hidden;
    }
  }
}

.TopBar,
.BottomBar {
  display: flex;
  flex-flow: row wrap;
  align-items: center;
  justify-content: space-between;
  color: white;
  min-height: 2.2rem;
  background-color: $horizontal-bar-color;
  .Filters,
  .Pagination {
    flex-wrap: wrap;
    display: flex;
    margin: 0.2em;
  }
  .FileInputButton,
  .info,
  button {
    line-height: 1;
    font-size: 0.8em;
    padding: 0.5em;
    margin: 0.1rem;
  }
  .info {
    flex: 1;
  }
  .FileInputButton,
  button {
    @extend %button;
    outline: none;
    color: white;
    background-color: $button-color;
    cursor: pointer;
    &:hover {
      background-color: scale_color(
        $button-color,
        $saturation: 100,
        $lightness: 10
      );
    }
    &.inactive {
      background-color: scale_color(
        $button-color,
        $lightness: -50,
        $saturation: -30
      );
      &:hover {
        background-color: scale_color(
          $button-color,
          $lightness: -30,
          $saturation: -10
        );
      }
    }
    &:disabled {
      box-shadow: none;
      background-color: scale_color(
        $button-color,
        $lightness: -30,
        $saturation: -90
      );
      color: #ccc;
      cursor: inherit;
    }
  }
}
