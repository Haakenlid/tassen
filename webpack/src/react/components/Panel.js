import React from 'react'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'

const Panel = ({ children }) =>
  <section className="Panel"  >
    { ...children }
  </section>

Panel.propTypes = {
  children: PropTypes.any.isRequired
}
const mapStateToProps = (state, ownProps) => ({

})
const mapDispatchToProps = (dispatch, ownProps) => ({

})
export default connect(mapStateToProps, mapDispatchToProps)(Panel)
