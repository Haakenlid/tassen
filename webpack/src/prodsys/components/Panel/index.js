import Panel from './Panel.js'
import './Panel.scss'
export default Panel
