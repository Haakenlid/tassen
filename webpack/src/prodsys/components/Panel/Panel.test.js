import Panel from './index'
import renderer from 'react-test-renderer'

describe('Panel', () => {
  test('renders correctly', () => {
    const tree = renderer.create(<Panel />).toJSON()
    expect(tree).toEqual(expect.any(Object))
    expect(tree).toMatchSnapshot()
  })
})
