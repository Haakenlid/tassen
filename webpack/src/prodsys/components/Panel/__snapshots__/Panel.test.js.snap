// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Panel renders correctly 1`] = `
<section
  className="Panel"
>
  <section
    className="content scroll"
  />
</section>
`;
